src/postgraphile/postgraphile.ts

```diff
@@ -114,41 +114,40 @@ export default function postgraphile(
   // must process them with `pluginHook` before we can rely on them.
   let incomingOptions: PostGraphileOptions;
 
-  // If the second argument is undefined, use defaults for both `schema` and
-  // `incomingOptions`.
-  if (typeof schemaOrOptions === 'undefined') {
-    schema = 'public';
-    incomingOptions = {};
-  }
   // If the second argument is a string or array, it is the schemas so set the
   // `schema` value and try to use the third argument (or a default) for
   // `incomingOptions`.
-  else if (typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
+  if (typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
     schema = schemaOrOptions;
     incomingOptions = maybeOptions || {};
```

Comment:
If we call `postgraphile(pool, null, {dynamicJson: true})` then I think the options object will be dropped - we don't want this. I think if we change the above to:

```js
if (!schemaOrOptions || typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
  schema = schemaOrOptions || 'public';
  incomingOptions = maybeOptions || {};
```

then this situation will be handled? What do you think?

Reply:
Handled 👍